Matveibos/hometask-4/my_hash.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'benchmark'
+
+# this class is Hash
+class MyHash
+  include Enumerable
+
+  attr_reader :pair_count, :size, :arr
+
+  LOAD_FACTOR = 0.5
+  SIZE = 10
+
+  def initialize
+    initialize_hash
+  end
+
+  def [](key)
+    ident = key.hash % @size
+    arr[ident].find { |k, _| k == key }&.last
+  end
+
+  def []=(key, value)
+    resize if (pair_count) > @size * LOAD_FACTOR
+    ident = key.hash % @size
+    pair = arr[ident].find { |k, _| k == key }
+    if pair.nil?
```

Comment:
Do we need so many checks here? Correct me, if I'm wrong, but we should reassign value if `pair` is present, or push key => value to our hash otherwise